File: app/src/main/java/com/test/xyz/demo/domain/interactor/project/ProjectInteractor.java
package com.test.xyz.demo.domain.interactor.project;

import com.test.xyz.demo.domain.model.github.GitHubRepo;

import java.util.List;

import io.reactivex.Observable;

public interface ProjectInteractor {
    Observable<List<GitHubRepo>> getProjectList(String userName);
    Observable<GitHubRepo> getProjectDetails(String userName, String projectID);
}

File: app/src/main/kotlin/com/test/xyz/demo/domain/interactor/project/ProjectInteractor.kt
package com.test.xyz.demo.domain.interactor.project

import com.test.xyz.demo.domain.model.github.GitHubRepo

import io.reactivex.Observable

interface ProjectInteractor {
    fun getProjectList(userName: String): Observable<List<GitHubRepo>>
    fun getProjectDetails(userName: String, projectID: String): Observable<GitHubRepo>
}
